csharp/Spotflow.EventHubConsumer/Program.cs

```diff
@@ -0,0 +1,142 @@
+    consumerGroup,
+    eventHubsConnectionString,
+    eventHubName);
+
+var partitionEventCount = new ConcurrentDictionary<string, int>();
+
+async Task processEventHandler(ProcessEventArgs args)
+{
+    try
+    {
+        // If the cancellation token is signaled, then the
+        // processor has been asked to stop.  It will invoke
+        // this handler with any events that were in flight;
+        // these will not be lost if not processed.
+        //
+        // It is up to the handler to decide whether to take
+        // action to process the event or to cancel immediately.
+
+        if (args.CancellationToken.IsCancellationRequested)
+        {
+            return;
+        }
+
+        var partition = args.Partition.PartitionId;
+        var eventBody = args.Data.EventBody.ToArray();
+        Console.WriteLine($"Event from partition {partition} with length {eventBody.Length}.");
+        var message = Encoding.UTF8.GetString(eventBody);
+        Console.WriteLine(message);
+
+        int eventsSinceLastCheckpoint = partitionEventCount.AddOrUpdate(
+            key: partition,
+            addValue: 1,
+            updateValueFactory: (_, currentCount) => currentCount + 1);
+
+        if (eventsSinceLastCheckpoint >= 50)
+        {
+            await args.UpdateCheckpointAsync();
+            partitionEventCount[partition] = 0;
+        }
+    }
+    catch
+    {
+        // It is very important that you always guard against
+        // exceptions in your handler code; the processor does
+        // not have enough understanding of your code to
+        // determine the correct action to take.  Any
+        // exceptions from your handlers go uncaught by
+        // the processor and will NOT be redirected to
+        // the error handler.
+    }
+}
+
+Task processErrorHandler(ProcessErrorEventArgs args)
+{
+    try
+    {
+        Console.WriteLine("Error in the EventProcessorClient");
+        Console.WriteLine($"\tOperation: {args.Operation}");
+        Console.WriteLine($"\tException: {args.Exception}");
+        Console.WriteLine("");
+    }
+    catch
+    {
+        // It is very important that you always guard against
+        // exceptions in your handler code; the processor does
+        // not have enough understanding of your code to
+        // determine the correct action to take.  Any
+        // exceptions from your handlers go uncaught by
+        // the processor and will NOT be handled in any
+        // way.
+    }
+
+    return Task.CompletedTask;
+}
+
+try
+{
+    using var cancellationSource = new CancellationTokenSource();
+    cancellationSource.CancelAfter(TimeSpan.FromSeconds(60));
+
+    processor.ProcessEventAsync += processEventHandler;
+    processor.ProcessErrorAsync += processErrorHandler;
+
+    try
+    {
+        await processor.StartProcessingAsync(cancellationSource.Token);
+        await Task.Delay(Timeout.Infinite, cancellationSource.Token);
+    }
+    catch (TaskCanceledException)
```

Comment:
```suggestion
    catch (OperationCanceledException) when (cts.IsCancellationRequested)
```

Otherwise, it is a bug.